fe1-web/parts/lao/organizer/eventCreation/events/CreateElection.tsx

```diff
@@ -0,0 +1,158 @@
+import ParagraphBlock from 'components/ParagraphBlock';
+import WideButtonView from 'components/WideButtonView';
+import { Timestamp } from 'model/objects';
+import TextBlock from 'components/TextBlock';
+import DropdownSelector from 'components/DropdownSelector';
+
+function dateToTimestamp(date: Date): Timestamp {
+  return new Timestamp(Math.floor(date.getTime() / 1000));
+}
+
+/**
+ * UI to create a Election Event
+ *
+ * TODO Send the Election event to the organization server when the confirm button is pressed
+ */
+const CreateElection = ({ route }: any) => {
+  const styles = route.params;
+  const navigation = useNavigation();
+  const initialStartDate = new Date();
+  const initialEndDate = new Date();
+  // Sets initial end date to 1 hour later than start date
+  initialEndDate.setHours(initialEndDate.getHours() + 1);
+
+  const [startDate, setStartDate] = useState(dateToTimestamp(initialStartDate));
+  const [endDate, setEndDate] = useState(dateToTimestamp(initialEndDate));
+
+  const [electionName, setElectionName] = useState('');
+  const [electionQuestion, setElectionQuestion] = useState('');
+  const [electionBallots, setElectionBallots] = useState(['']);
+  const [electionBallotCounter, setElectionBallotCounter] = useState(2);
+  const votingMethods = [STRINGS.election_method_Plurality];
+  const [votingMethod, setVotingMethod] = useState(votingMethods[0]);
+  const ballotOptionsUIComponents = [];
+
+  const buildDatePickerWeb = () => {
+    const startTime = new Date(0);
+    const endTime = new Date(1);
+    startTime.setUTCSeconds(startDate.valueOf());
+    endTime.setUTCSeconds(endDate.valueOf());
+
+    return (
+      <View style={styles.view}>
+        <ParagraphBlock text={STRINGS.election_create_start_time} />
+        <DatePicker
+          selected={startTime}
+          onChange={(date: Date) => setStartDate(dateToTimestamp(date))}
+        />
+        <ParagraphBlock text={STRINGS.election_create_finish_time} />
+        <DatePicker
+          selected={endTime}
+          onChange={(date: Date) => setEndDate(dateToTimestamp(date))}
+        />
+      </View>
+    );
+  };
+
+  // Confirm button only clickable when the Name, Question and 2 Ballot options have values
+  const buttonsVisibility: boolean = (electionQuestion !== '' && electionName !== ''
+    && electionBallots.length >= 2 && electionBallots[0] !== '' && electionBallots[1] !== '');
+
+  // Makes sure you can't remove ballots when there are only 2 options
+  const removeButtonVisibility: boolean = (electionBallotCounter > 2);
+
+  const onAddBallotPress = () => {
+    setElectionBallotCounter((prevCount) => prevCount + 1);
+  };
+
+  const onRemoveBallotPress = () => {
+    // decrements counter
+    setElectionBallotCounter((prevCount) => prevCount - 1);
+    // removes value from ballot array
+    const newArr = [...electionBallots];
+    newArr.splice(electionBallotCounter - 1, 1);
+    setElectionBallots(newArr);
+  };
```

Comment:
Here this arrow function should get the **ballot option index** as a parameter and update the state according to the index. Could look something like:


```
const onRemoveBallotPress = (ballotId) => {

        setElectionBallotCounter((prevCount) => prevCount - 1);
        const newArr = electionBallors.filter(b => b.id !== ballotId);
        ......
        setElectionBallots(newArr);

```

Reply:
I tend to agree, but I see this was resolved as well, was there some discussion between you two? :)

Reply:
This is now implemented